Changes required by Zig 0.14.0 but applicable in Zig 0.13.0

batiati opened the pull request:
In preparation for #2705.

batiati committed:
Prefer the `Scalar` variant of std.mem.indexOf, lastIndexOf, tokenize and split.

batiati committed:
Fix field names shadowed by functions.

batiati committed:
Fix invalid implicity cast from `[][:0]const u8` to `[][]const u8`.
Ref https://github.com/ziglang/zig/issues/23016

batiati committed:
Fix Vortex Storage == void usage.

matklad commented on src/vsr/message_header.zig:
```diff
@@ -721,7 +721,7 @@ pub const Header = extern struct {
         }
 
-        pub fn reserved(cluster: u128, slot: u64) Prepare {
+        pub fn reserved_operation(cluster: u128, slot: u64) Prepare {
```
Not a huge fan of this name, but nothing better comes to mind. Maybe `reserved_slot`? Anyway, shouldn't be a problem to rename later!

matklad commented on src/io/windows.zig:
```diff
@@ -1394,8 +1394,19 @@ pub const IO = struct {
+                .library_name = "kernel32",
+                .name = "SetEndOfFile",
+            },
+        );
```
I am curious here, what's the context of these being removed from std? As in, should we perhaps be using something else?

batiati replied:
This:
https://github.com/ziglang/zig/commit/3095e83d1162081ba54ed73c59810bfe0fb3f025

batiati replied:
> should we perhaps be using something else?

I've added a `TODO(zig)` for this one (planning to take a look after the migration)
https://github.com/tigerbeetle/tigerbeetle/pull/2814/commits/f240d7dcdc3622b09de6bdbd87f914b8b876d83b

matklad commented on src/io/windows.zig:
```diff
@@ -1394,8 +1394,19 @@ pub const IO = struct {
         }
 
+        // Declaring the function with an alternative name because `CamelCase` functions are
+        // by convention, used for building generic types.
```
TBH, I'd remove the comment --- it is not particularly relevant to what we are doing here so it distracts the reader. If anyone tries to rename it to be more idiomatic, they'll learn why it is the way it is :)

matklad replied:
But that would be great tidbit to include in the commit message: https://mislav.net/2014/02/hidden-documentation/

matklad commented on src/vsr/journal.zig:
```diff
@@ -1525,7 +1525,8 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
                     assert(
-                        header.?.checksum == Header.Prepare.reserved_operation(cluster, slot.index).checksum,
+                        header.?.checksum ==
+                            Header.Prepare.reserved_operation(cluster, slot.index).checksum,
```
Oh, maybe `Header.Prepare.reserve(cluster, slot.index)`? As a verb?

batiati replied:
`reserve` as a verb sounds more like an acquire/release mechanism.

maybe `as_reserved()`, `from_reserved()`, or the other way around, renaming the field `reserved` to something else like `reserved_bytes` or `padding`.

batiati committed:
Fix field names shadowed by functions.

batiati committed:
[Windows] Deprecated APIS
- Redeclare kernel32 functions removed from std.
  Declaring the function with an alternative name because `CamelCase` functions are by convention, used for building generic types.
- Use `utf8ToUtf16LeAlloc` instead of the deprecated `utf8ToUtf16LeWithNull`.
- Add TODO note on realtime_windows.
